lib/compat/wordpress-6.0/class-wp-webfonts-provider-local.php

```diff
@@ -227,9 +227,9 @@ private function compile_src( $font_family, array $value ) {
 		$src = "local($font_family)";
 
 		foreach ( $value as $item ) {
-
-			if ( 0 === strpos( $item['url'], get_site_url() ) ) {
-				$item['url'] = wp_make_link_relative( $item['url'] );
+			if ( 0 === strpos( $item['url'], 'file:./' ) ) {
+				$absolute_path_to_url = get_stylesheet_directory_uri() . '/' . str_replace( 'file:./', '', $item['url'] );
+				$item['url']          = wp_make_link_relative( $absolute_path_to_url );
 			}
```

Comment:
`'file:./'` is specific to `theme.json` as the absolute path cannot be set in it. What about moving this code to the theme.json parser? Why?
- To keep the Webfonts API source agnostic.
- To bundle the specific details related to `theme.json` in its parser or handler.
- As the web fonts configuration parameters are being parsed first before passing to this API, that parser could also do the work it needs to make the font file URL absolute.